### DISCUSSION TITLE
Is there any way of verifying if vmware tools are installed in vm?

### DISCUSSION THREAD

**Discussion by mmfreitas** [bb-Ricardo/netbox-sync#300] [Q&A]
I did some test runs in my environment and noticed that IP addresses that are already assigned in netbox get unassigned from the interface of th virtual machine.
This is due to the VM not having the vmware tools installed.

So my question is if there is a way of skipping the IP Address assignement in the VM's that do not have vmware tools installed?

**Comment by bb-Ricardo** [comment]
Hey,

I'm curious, the assigned IPs get removed if they were assigned previous to the first sync?

But it is also not that easy. Even if the tools are installed in an older version they sometimes don't expose IP addresses.

It could be an option to disable the IP sync if vmware tools are not installed.

**Comment by mmfreitas** [comment]
Hey,

Sorry for the late response.

Yes, the IP's were assigned previous to the first sync.

With further testing I've noticed that it is not only VM's which don't have vmware tools installed but also virtual machines which are offline that get the IP's which are assigned to it's interfaces unassigned and the primary_ip also removed.

Having that option to disable IP sync when vmware tools are not detected or when the vm is offline would be great to avoid this data which is configured on the virtual machines, but not "visible" to netbox-sync.

**Comment by mmfreitas** [comment]
Should I open a feature request for this?

**Comment by bb-Ricardo** [comment]
mmhm, just checked. NetBox-Sync removes the IP address if the VM is offline, that is correct.

The IP is not used und not actively assigned to the interface, therefore it gets removed.

**Comment by bb-Ricardo** [comment]
I could probably add a config option to disable the IP address handling if VMWare tools are not installed. Would this help?

**Reply by mmfreitas** [reply]
> I could probably add a config option to disable the IP address handling if VMWare tools are not installed. Would this help?

Yes, I think that would be sufficient to resolve this issue.
Thanks a lot!

**Reply by bb-Ricardo** [reply]
Would you be able to create a Feature request out of it?

**Reply by mmfreitas** [reply]
I have opened #326.
